packages/expo-updates/android/src/main/java/expo/modules/updates/db/dao/AssetDao.kt

```diff
@@ -89,10 +89,24 @@ abstract class AssetDao {
 
   fun mergeAndUpdateAsset(existingEntity: AssetEntity, newEntity: AssetEntity) {
     // if the existing entry came from an embedded manifest, it may not have a URL in the database
-    if (newEntity.url != null && existingEntity.url == null) {
+    var shouldUpdate = false
+    if ((newEntity.url != null && existingEntity.url == null) || newEntity.url != existingEntity.url) {
```

Comment:
Agree with @ide 's suggestion to separate null checks from nonnull equality checks -- we don't, for example, want to update the URL if `newEntity.url` is null but `existingEntity.url` is nonnull. And same with the extraRequestHeaders, I think. (This can happen if newEntity is coming from an embedded manifest, since embedded manifests don't include any information about how to download their assets from a remote server.)